CMake: INTERFACE_LIBRARY targets may only have whitelisted properties

**Describe the bug**
simdjson 4.0.0 uses target property that isn't whitelisted and leads to failure of CMake generation for older versions.

```
СMake Error at cmake/developer-options.cmake:7 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "POSITION_INDEPENDENT_CODE" is not allowed.
```

**To Reproduce**
`cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DSIMDJSON_ENABLE_THREADS=OFF -DSIMDJSON_DEVELOPER_MODE=OFF /usr/src`

<details><summary>Click to expand log</summary>

```

-- The CXX compiler identification is GNU 11.4.0
-- The C compiler identification is GNU 11.4.0
-- Check for working CXX compiler: /usr/local/bin/c++
-- Check for working CXX compiler: /usr/local/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/local/bin/cc
-- Check for working C compiler: /usr/local/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
CMake Warning at CMakeLists.txt:50 (message):
  SIMDJSON_STATIC_REFLECTION is disabled because your CMake version is below
  3.25


CMake Error at cmake/developer-options.cmake:7 (set_target_properties):
  INTERFACE_LIBRARY targets may only have whitelisted properties.  The
  property "POSITION_INDEPENDENT_CODE" is not allowed.
Call Stack (most recent call first):
  CMakeLists.txt:76 (include)


-- Looking for fork
-- Looking for fork - found
-- Looking for wait
-- Looking for wait - found
-- Adding -Og to compile flag
-- Building only the library. Advanced users and contributors may want to turn SIMDJSON_DEVELOPER_MODE to ON, e.g., via -D SIMDJSON_DEVELOPER_MODE=ON.
-- Configuring incomplete, errors occurred!

```
</details>

**simdjson release**
4.0.0

**Configuration (please complete the following information if relevant)**
- OS: Ubuntu 16.04
- Compiler: gcc 11.4.0
- CMake 3.15.7

**Indicate whether you are willing or able to provide a bug fix as a pull request**
I'll prepare PR with tiny change that fixes this.

--- a/cmake/developer-options.cmake
+++ b/cmake/developer-options.cmake
@@ -4,7 +4,7 @@
 add_library(simdjson-internal-flags INTERFACE)
 if(NOT DEFINED CMAKE_POSITION_INDEPENDENT_CODE)
   # We default to ON for all targets, so that we can use the library in shared libraries.
-  set_target_properties(simdjson-internal-flags PROPERTIES POSITION_INDEPENDENT_CODE ON)
+  set_target_properties(simdjson-internal-flags PROPERTIES INTERFACE_POSITION_INDEPENDENT_CODE ON)
 endif(NOT DEFINED CMAKE_POSITION_INDEPENDENT_CODE)
 
 option(SIMDJSON_CHECK_EOF "Check for the end of the input buffer. The setting is unnecessary since we require padding of the inputs. You should expect tests to fail with this option turned on." OFF)

